ax/modelbridge/modelbridge_utils.py

```diff
@@ -189,61 +192,70 @@ def extract_robust_digest(
                 "All distributional parameters must be included in `param_names`."
             )
         dist_idcs[p_name] = param_names.index(p_name)
-    distributions = search_space.parameter_distributions
     num_samples = search_space.num_samples
-    multiplicative = distributions[0].multiplicative
-    if search_space.is_environmental:
-        num_non_dist_params = len(param_names) - len(dist_params)
-        if set(dist_idcs.values()) != set(range(num_non_dist_params, len(param_names))):
+    if len(env_vars) > 0:
+        num_non_env_vars = len(param_names) - len(env_vars)
+        env_idcs = set(idx for p, idx in dist_idcs.items() if p in env_vars)
+        if env_idcs != set(range(num_non_env_vars, len(param_names))):
             raise RuntimeError(
                 "Environmental variables must be last entries in `param_names`. "
                 "Otherwise, `AppendFeatures` will not work."
             )
         # NOTE: Extracting it from `param_names` in case the ordering is different.
-        environmental_variables = param_names[num_non_dist_params:]
+        environmental_variables = param_names[num_non_env_vars:]
 
-        def get_samples() -> np.ndarray:
+        def environmental_sampler() -> np.ndarray:
             """Get samples from the environmental distributions.
 
             Samples have the same dimension as the number of environmental variables.
             The samples of an environmental variable appears in the same order it is
             in `param_names`.
             """
-            samples = np.zeros((num_samples, len(dist_params)))
-            for dist in distributions:
+            samples = np.zeros((num_samples, len(env_vars)))
+            # pyre-ignore [16]
+            for dist in search_space._environmental_distributions:
                 dist_samples = dist.distribution.rvs(num_samples).reshape(
                     num_samples, -1
                 )
                 for i, p_name in enumerate(dist.parameters):
-                    target_idx = dist_idcs[p_name] - num_non_dist_params
+                    target_idx = dist_idcs[p_name] - num_non_env_vars
                     samples[:, target_idx] = dist_samples[:, i]
             return samples
 
     else:
+        environmental_sampler = None
         environmental_variables = []
 
-        def get_samples() -> np.ndarray:
+    if len(pert_params) > 0:
+        constructor = np.ones if search_space.multiplicative else np.zeros
+
+        def distribution_sampler() -> np.ndarray:
```

Comment:
"distribution" as a term is a bit unclear - we're always sampling from distributions. Assuming this is about (possibly degenerate) parameter perturbations how about the following (assuming this doesn't cause a bunch of backward compatibility issues if we also change it in the `RobustSearchSpace`)?
```suggestion
        def sample_param_perturbations() -> np.ndarray:
```

Reply:
yeah, I considered renaming this to `perturbation_sampler` but left it as is mostly due to laziness. This stuff is new enough and is still not supported throughout the modeling stack (except for my stack of diffs), so it should be ok to break BC.